Release notes for OpenNMS Horizon 31.0.9 (#6225)
OpenNMS/opennms: docs/modules/releasenotes/pages/changelog.adoc

[[release-31-changelog]]
= Changelog
[[releasenotes-changelog-31.0.9]]

==== Release 31.0.9

Release 31.0.9 contains one CVE-related security fix, a generous helping of other bug fixes, and a several small enhancements intended to improve supportability.

The codename for Horizon 31.0.9 is https://wikipedia.org/wiki/$$Ballokume$$[_Ballokume_].

===== Breaking changes

> * This release has moved to a newer major version of Spring Security to address a number of CVEs, which necessitated changes to the `$OPENNMS_HOME/jetty-webapps/opennms/WEB-INF/applicationContext-spring-security.xml` file, so if you have modified this file in your installs, be sure to note your changes so you can re-apply them to the updated version.

Reviewer: ```suggestion
* This release has moved to a newer major version of Spring Security to address a number of CVEs, which necessitated changes to the `$OPENNMS_HOME/jetty-webapps/opennms/WEB-INF/applicationContext-spring-security.xml` file, so if you have modified this file in your installs, be sure to note your changes so you can reapply them to the updated version.
```

===== Enhancement

* Codify code copyright conventions and guidelines (Issue https://issues.opennms.org/browse/NMS-13908[NMS-13908])
* Add diagnostic commands to Karaf shell for various internal schedulers (Issue https://issues.opennms.org/browse/NMS-14526[NMS-14526])
* Node Properties REST endpoint doesn't include asset location data (Issue https://issues.opennms.org/browse/NMS-14785[NMS-14785])
* Add a method for finding and clearing alarms by TTicketID to OPA's AlarmDAO (Issue https://issues.opennms.org/browse/NMS-15439[NMS-15439])
* Upgrade Spring Security (Issue https://issues.opennms.org/browse/NMS-15506[NMS-15506])
* Simplify the installation docs (Issue https://issues.opennms.org/browse/NMS-15518[NMS-15518])
* Docs: Add info about XSLT to XmlCollector (Issue https://issues.opennms.org/browse/NMS-15693[NMS-15693])
* Doc: Update DNS provisioning import adapter docs (Issue https://issues.opennms.org/browse/NMS-15694[NMS-15694])

===== Task

* Remove unsupported configuration from documentation on Cortex Plugin (Issue https://issues.opennms.org/browse/NMS-14969[NMS-14969])
* Multiple CVEs for Axis 1.4 (Issue https://issues.opennms.org/browse/NMS-15061[NMS-15061])

> ===== Bug

Reviewer: ```suggestion
=== Bugs
```

* Fixing typo for event uei.opennms.org/internal/schedOutagesChanged (Issue https://issues.opennms.org/browse/NMS-15421[NMS-15421])
* Sentinels need local copy of thresholding config. (Issue https://issues.opennms.org/browse/NMS-15422[NMS-15422])
* Event Datetime element parsing changed between M2018 and M2021 (Issue https://issues.opennms.org/browse/NMS-15471[NMS-15471])
* install script checks for equality of myuser and RUNAS before sourcing opennms.conf (Issue https://issues.opennms.org/browse/NMS-15610[NMS-15610])
* timeout is using DEFAULT_TIMEOUT value instead of the value from properties file when no -t option is specified (Issue https://issues.opennms.org/browse/NMS-15664[NMS-15664])
* Stop packaging activemq-web-console.war (Issue https://issues.opennms.org/browse/NMS-15686[NMS-15686])
* Database deadlock caused by JdbcFilterDao (Issue https://issues.opennms.org/browse/NMS-15696[NMS-15696])
* Karaf SSH locks up if connections are terminated improperly (Issue https://issues.opennms.org/browse/NMS-15714[NMS-15714])

> ===== Story

Reviewer: ```suggestion
=== Story
```
